gtk/src/file_filter_info.rs

```diff
@@ -0,0 +1,71 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::translate::*;
+use libc::c_char;
+
+use crate::FileFilterFlags;
+
+#[derive(Clone)]
+pub struct FileFilterInfo {
+    pub contains: FileFilterFlags,
+    pub filename: Option<String>,
+    pub uri: Option<String>,
+    pub display_name: Option<String>,
+    pub mime_type: Option<String>,
+}
+
+#[doc(hidden)]
+impl From<*const ffi::GtkFileFilterInfo> for FileFilterInfo {
```

Comment:
This is not safe so shouldn't be marked as such :)

Why don't you implement the `FromGlibPtr*` and `ToGlibPtr*` traits here instead?

Reply:
Because I don't entirely know what I'm doing :-P

Reply:
No problem :) Just ask if you have questions, this is all not entirely obvious